pkg/controller/node/nodecontroller.go

```diff
@@ -483,9 +483,24 @@ func (nc *NodeController) monitorNodeStatus() error {
 						continue
 					}
 					if remaining {
-						// queue eviction of the pods on the node
+						// Immediately evict pods (skip rate-limited evictor)
 						glog.V(2).Infof("Deleting node %s is delayed while pods are evicted", node.Name)
-						nc.evictPods(node.Name)
+						go func(nodeName string) {
+							nc.evictorLock.Lock()
+							defer nc.evictorLock.Unlock()
+							remaining, err := nc.deletePods(nodeName)
+							if err != nil {
+								glog.Errorf("Unable to evict pods from node %s: %v", nodeName, err)
+								nc.podEvictor.Add(nodeName)
+								return
+							}
+							if !remaining {
+								return
+							}
+							// Queue for termination.
+							nc.terminationEvictor.Add(nodeName)
```

Comment:
I don't think we should rate limit it at all. IIUC this if branch is responsible for deleting stuff from etcd after node is gone. Hence no sigterms or sigkills will be emitted. You can reuse deletePods/terminatePods functions, but you don't need to rate limit any of them.